Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -312,14 +301,62 @@ public class ResourcesProjectMapper: ProjectMapping { // swiftlint:disable:this
         // MARK: - Swift Bundle Accessor for Frameworks
         private class BundleFinder {}
         extension Foundation.Bundle {
-        /// Since \(target.name) is a \(
-            target
-                .product
-        ), the bundle for classes within this module can be used directly.
+        /// Since \(target.name) is a \(target.product), \
+        the bundle for classes within this module can be used directly.
         static let module = Bundle(for: BundleFinder.self)
         }
         """
     }
+
+    private static func swiftStaticFrameworkBundleAccessorString(for target: Target) -> String {
+        """
+        // MARK: - Swift Bundle Accessor for Frameworks
+        extension Foundation.Bundle {
+        /// Since \(target.name) is a \(target.product), \
+        a cut down framework is embedded, with all the resources but only a stub Mach-O image.
+        static let module: Bundle = {
+            var candidates = [Bundle.main.privateFrameworksURL]
+            \(iterateOverCandidatesString(appendingPath: "\"\(target.name).framework\""))
+            fatalError("unable to find \(target.product) \\"\(target.name).framework\\"")
+        }()
+        }
+        \(appendingPathToURLString())
+        """
+    }
+
+    private static func iterateOverCandidatesString(appendingPath: String) -> String {
+        """
+        // This is a fix to make unit tests work with bundled resources.
+            // Making this change allows unit tests to search one directory up for the framework.
+            // More context can be found in this PR: https://github.com/tuist/tuist/pull/6895
```

Comment:
The padding looks off here.

Reply:
yes, but generated code looks fine. didn't get why such padding is needed here

Reply:
You should be able to pad it to the left to make things align, otherwise, this looks weird, although I understand it might not have any effect on the final generated file.